### PR TITLE
Better error message for DataFrame.hist() without numerical columns (…

### LINKED ISSUE #10444: Better error message when using DataFrame.hist() without numerical columns
pandas version: 0.16.2
matplotlib version 1.4.3 (and produced different error message on older version)

``` python
import numpy as np
import pandas as pd

df = pd.DataFrame(np.random.rand(10,2))
df_o = df.astype(np.object)
df_o.hist()
```

```
ValueError                                Traceback (most recent call last)
<ipython-input-1-26253737011d> in <module>()
      4 df = pd.DataFrame(np.random.rand(10,2))
      5 df_o = df.astype(np.object)
----> 6 df_o.hist()

/usr/local/lib/python2.7/dist-packages/pandas/tools/plotting.pyc in hist_frame(data, column, by, grid, xlabelsize, xrot, ylabelsize, yrot, ax, sharex, sharey, figsize, layout, bins, **kwds)
   2764     fig, axes = _subplots(naxes=naxes, ax=ax, squeeze=False,
   2765                           sharex=sharex, sharey=sharey, figsize=figsize,
-> 2766                           layout=layout)
   2767     _axes = _flatten(axes)
   2768 

/usr/local/lib/python2.7/dist-packages/pandas/tools/plotting.pyc in _subplots(naxes, sharex, sharey, squeeze, subplot_kw, ax, layout, layout_type, **fig_kw)
   3244 
   3245     # Create first subplot separately, so we can share it if requested
-> 3246     ax0 = fig.add_subplot(nrows, ncols, 1, **subplot_kw)
   3247 
   3248     if sharex:

/usr/local/lib/python2.7/dist-packages/matplotlib/figure.pyc in add_subplot(self, *args, **kwargs)
    962                     self._axstack.remove(ax)
    963 
--> 964             a = subplot_class_factory(projection_class)(self, *args, **kwargs)
    965 
    966         self._axstack.add(key, a)

/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_subplots.pyc in __init__(self, fig, *args, **kwargs)
     62                     raise ValueError(
     63                         "num must be 0 <= num <= {maxn}, not {num}".format(
---> 64                             maxn=rows*cols, num=num))
     65                 if num == 0:
     66                     warnings.warn("The use of 0 (which ends up being the "

ValueError: num must be 0 <= num <= 0, not 1
```

### PR DESCRIPTION
**Pull request by matsmaiwald**
Closes #10444

Added simple check for non-zero number of numeric columns plus suggested error message in case the check fails. 

Happy to make any adjustments this if desired.

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -2426,6 +2426,10 @@ def hist_frame(data, column=None, by=None, grid=True, xlabelsize=None,
     data = data._get_numeric_data()
     naxes = len(data.columns)
 
+    if naxes == 0:
```
Comment (simonjayhawkins) [inline pandas/plotting/_core.py]:
```suggestion
    if not naxes:
```

**Comment by simonjayhawkins** [comment]
@matsmaiwald 

thanks for the PR.

the CI failure looks unrelated to your change.

can you add a test, perhaps using the code sample in the issue as the test case?

**Comment by matsmaiwald** [comment]
Thanks for the code review @simonjayhawkins.

I just added a test in the way you suggested.

File: pandas/tests/plotting/test_hist_method.py [hunk 2]
```diff
@@ -209,6 +209,16 @@ def test_hist_df_legacy(self):
         with pytest.raises(AttributeError):
             ser.hist(foo='bar')
 
+    @pytest.mark.slow
+    def test_hist_non_numerical(self):
```
Comment (simonjayhawkins) [inline pandas/tests/plotting/test_hist_method.py]:
```suggestion
    def test_hist_non_numerical_raises(self):
```

File: pandas/tests/plotting/test_hist_method.py [hunk 3]
```diff
@@ -209,6 +209,16 @@ def test_hist_df_legacy(self):
         with pytest.raises(AttributeError):
             ser.hist(foo='bar')
 
+    @pytest.mark.slow
+    def test_hist_non_numerical_raises(self):
+        # check non-numerical df raises appropriate exception and error message
```
Comment (simonjayhawkins) [inline pandas/tests/plotting/test_hist_method.py]:
```suggestion
        # gh-10444
```

**Comment by simonjayhawkins** [comment]
lgtm @datapythonista @TomAugspurger

File: pandas/plotting/_core.py (hunk 1 above)
Comment (gfyoung) [reply pandas/plotting/_core.py]: I prefer how it is.  This is more explicit.

**Review by gfyoung — APPROVED** [review]

**Comment by datapythonista** [comment]
lgtm, not sure about marking the test as slow, is it really slow?

Also, if this can wait until #26414 that would make my life a bit easier (will be much easier to resolve the conflict here than there).

**Comment by simonjayhawkins** [comment]
> lgtm, not sure about marking the test as slow, is it really slow?

that had crossed my mind, but all the tests in this module are marked as slow, so was thinking that it would be better to mark as slow at the module level as a follow-up at some point.

**Comment by jreback** [comment]
thanks @matsmaiwald

**Pull request merged**